src/main/java/dao/Implementations/UserDaoHibernateImpl.java

```diff
@@ -0,0 +1,68 @@
+package dao.Implementations;
+
+import dao.Interfaces.UserDao;
+import Entity.User;
+import org.hibernate.Criteria;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import utils.HibernateSessionFactoryUtil;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+public class UserDaoHibernateImpl implements UserDao {
+
+    private static UserDaoHibernateImpl userDAO;
+    private final SessionFactory sessionFactory = HibernateSessionFactoryUtil.getSessionFactory();
+
+    public static UserDaoHibernateImpl instance() {
+        if (userDAO == null) {
+            userDAO = new UserDaoHibernateImpl();
+        }
+        return userDAO;
+    }
+
+    private UserDaoHibernateImpl() {
+    }
+
+    @Override
+    public boolean addUser(String login, String password) {
+        if ((Objects.isNull(login)) || Objects.isNull(password)
+                || (login.isEmpty()) || (password.isEmpty())) {
+            return false;
+        }
+        if (getUserByLogin(login) == null) {
+            Session session = sessionFactory.openSession();
+            session.beginTransaction();
+            session.save(new User(login, password));
+            session.getTransaction().commit();
+            session.close();
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public User getUserByLogin(String login) {
+        Session session = sessionFactory.openSession();
+        Query query = session.createQuery("from User where login=:userLogin");
```

Comment:
better make query as constant
```suggestion
        Query query = session.createQuery("FROM User WHERE login=:userLogin");
```